### PR TITLE
Add support for multiple backends

### PR DESCRIPTION
**Pull request by ksooo**
See https://github.com/xbmc/xbmc/pull/20860

### CODE REVIEW

**Comment by ksooo** [comment]
BTW, this fixes #361

File: src/tvheadend/utilities/SettingsMigration.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  if (mig.Changed())
+  {
+    // Set a title for the new instance settings
+    std::string title;
+    target.CheckInstanceSettingString("host", title);
+    if (title.empty())
+      title = "Migrated Add-on Config";
+
+    target.SetInstanceSettingString("kodi_addon_instance_name", title);
```
Comment (Jalle19) [inline src/tvheadend/utilities/SettingsMigration.cpp:83]: Would perhaps be better to use a dedicated setting for this, e.g. `setting_migration_done`, that way we don't run into unexpected issues if for whatever reason the `kodi_addon_instance_name` setting is later renamed or removed.

Comment (ksooo) [reply src/tvheadend/utilities/SettingsMigration.cpp:83]: Yes, definitely better. I will change that.

Comment (ksooo) [reply src/tvheadend/utilities/SettingsMigration.cpp:83]: On second thought, I think we actually don't need another setting. The only thing that might go wrong is that if somebody removes the instance title by hand from intsance-settings-1.xml (Kodi GUI prevents this), we migrate the settings for this instance a second time, overwriting the previously user-set values with the old values from settings.xml. Not nice, but if users manipulate files in a way they are not supposed to to, well ...

Comment (Jalle19) [reply src/tvheadend/utilities/SettingsMigration.cpp:83]: Alright, sure!

**Pull request merged**